app/models/user.rb

```diff
@@ -56,11 +56,13 @@ def from_omniauth(auth)
       # Provider is the customer name if in loadbalanced config mode
       provider = auth['provider'] == "bn_launcher" ? auth['info']['customer'] : auth['provider']
       find_or_initialize_by(social_uid: auth['uid'], provider: provider).tap do |u|
-        u.name = auth_name(auth) unless u.name
-        u.username = auth_username(auth) unless u.username
+        u.name = auth_name(auth)
+        u.username = auth_username(auth)
         u.email = auth_email(auth)
         u.image = auth_image(auth)
         u.email_verified = true
+        roles = auth_roles(auth)
```

Comment:
This should be done in a different way as it is specific to auth0. What if no roles are included, or roles are included in a different attribute (for other providers)